idb/postgres/postgres_migrations.go

```diff
@@ -1936,3 +1938,104 @@ func m11AssetHoldingFrozen(db *IndexerDb, state *MigrationState) error {
 
 	return nil
 }
+
+type m12Account struct {
+	address  sdk_types.Address
+	closedAt uint64 // the round when the account was last closed
+}
+
+func m12GetAccounts(db *sql.DB) ([]m12Account, error) {
+	query := "SELECT addr, created_at FROM account WHERE closed_at IS NOT NULL AND deleted = false " +
+		"AND account_data IS NOT NULL"
+	rows, err := db.Query(query)
+	if err != nil {
+		return []m12Account{}, err
+	}
+	defer rows.Close()
+
+	var res []m12Account
+
+	for rows.Next() {
+		var addrBytes []byte
+		var closedAt sql.NullInt64
+
+		err = rows.Scan(&addrBytes, &closedAt)
+		if err != nil {
+			return []m12Account{}, err
+		}
+
+		var addr sdk_types.Address
+		copy(addr[:], addrBytes)
+		res = append(res, m12Account{addr, uint64(closedAt.Int64)})
+	}
+	if err := rows.Err(); err != nil {
+		return []m12Account{}, err
+	}
+
+	return res, nil
+}
+
+func m12FixAuthAddr(db *sql.DB, account m12Account) error {
```

Comment:
Not sure it's worth it, but I think we can do this for all accounts with a query something like

```
UPDATE account SET account_data - 'spend' WHERE addr = $1 AND 0 < (SELECT count(*) FROM txn JOIN txn_participation ON round=round AND intra=intra WHERE round > $2 AND (txn -> 'txn' ->> 'rekey') IS NOT NULL AND addr = $1 LIMIT 1)
```